File: view/src/view_custom/java/com/sample/view/VideoTrimmerAdapter.java
package com.sample.view;

import android.content.Context;
import android.graphics.Bitmap;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;


public class VideoTrimmerAdapter extends RecyclerView.Adapter {
    private Context context;
    private float itemWidth;
    private LayoutInflater mInflater;
    private List<Bitmap> mBitmaps = new ArrayList<>();

    public VideoTrimmerAdapter(Context context) {
        this.context = context;
        this.mInflater = LayoutInflater.from(context);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new TrimmerViewHolder(mInflater.inflate(R.layout.video_thumb_item_layout, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        ((TrimmerViewHolder) holder).thumbImageView.setImageBitmap(mBitmaps.get(position));
    }

    @Override
    public int getItemCount() {
        return mBitmaps.size();
    }

    public void clearBitmaps() {
        if (null != mBitmaps) {
            mBitmaps.clear();
        }
    }
    public void addBitmaps(Bitmap bitmap) {
        if (null != mBitmaps) {
            mBitmaps.add(bitmap);
            notifyDataSetChanged();
        }
    }

    public void setItemWidth(float itemWidth) {
        this.itemWidth = itemWidth;
    }

    private final class TrimmerViewHolder extends RecyclerView.ViewHolder {
        ImageView thumbImageView;

        TrimmerViewHolder(View itemView) {
            super(itemView);
            thumbImageView = itemView.findViewById(R.id.thumb);
            LinearLayout.LayoutParams layoutParams = (LinearLayout.LayoutParams) thumbImageView.getLayoutParams();
            layoutParams.width = (int) itemWidth;
            thumbImageView.setLayoutParams(layoutParams);
        }
    }
}

File: app/src/view_custom/java/com/sample/view/VideoTrimmerAdapter.kt
package com.sample.view

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

/**
 *  @author mzp
 *  date : 2020/11/4
 *  desc :
 */
class VideoTrimmerAdapter(private var context: Context): RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {

        val view = LayoutInflater.from(context).inflate(if (viewType == 1) R.layout.item_empty_adapter else R.layout.item_adapter, parent, false)
        return MyViewholder(view)
    }

    override fun getItemCount(): Int {
        return 12
    }

    override fun getItemViewType(position: Int): Int {
        if (position == 0 || position == 11) {
            return 1
        }
        return 0
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {

    }

    class MyViewholder(view: View) : RecyclerView.ViewHolder(view) {

    }
}